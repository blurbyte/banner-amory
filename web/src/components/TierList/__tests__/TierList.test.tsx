import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { MockedProvider } from 'react-apollo/test-utils';
import * as delay from 'delay';

import { TierList } from '../index';
import { itemsMock } from './mocks';

beforeEach(cleanup);

test('renders loading state initially', () => {
  const { container } = render(
    <MockedProvider mocks={itemsMock} addTypename={false}>
      <TierList />
    </MockedProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});

test('renders loaded list of items', async () => {
  const { container } = render(
    <MockedProvider mocks={itemsMock} addTypename={false}>
      <TierList />
    </MockedProvider>
  );

  await delay(0);

  expect(container.firstChild).toMatchSnapshot();
});
