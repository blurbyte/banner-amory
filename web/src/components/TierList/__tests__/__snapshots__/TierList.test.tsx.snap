// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loaded list of items 1`] = `
<article
  class="sc-cSHVUG bcwYBM sc-jzJRlG hOHjtX"
>
  <h1
    class="sc-jTzLTM eAGGiO"
  >
    Survival Mode Tier List
  </h1>
  <ul
    class="sc-VigVT dFmCvC"
  >
    <li
      class="sc-htpNat hkrMkj"
    >
      <a
        class="sc-bxivhb jgmcrE"
        href="/item/ariana-grande"
      >
        <img
          class="sc-bdVaJa hpZUqN"
          src="undefinedariana-grande.png"
        />
        <p
          class="sc-iwsKbI jrVOTo sc-EHOje RRuYV"
        >
          10
        </p>
      </a>
    </li>
    <li
      class="sc-htpNat hkrMkj"
    >
      <a
        class="sc-bxivhb jgmcrE"
        href="/item/taylor-swift"
      >
        <img
          class="sc-bdVaJa hpZUqN"
          src="undefinedtaylor-swift.png"
        />
        <p
          class="sc-iwsKbI jrVOTo sc-EHOje RRuYV"
        >
          1
        </p>
      </a>
    </li>
  </ul>
</article>
`;

exports[`renders loading state initially 1`] = `
<article
  class="sc-cSHVUG bcwYBM sc-jzJRlG hOHjtX"
>
  <h1
    class="sc-jTzLTM eAGGiO"
  >
    Survival Mode Tier List
  </h1>
</article>
`;
